loginpass/vk.py

```diff
@@ -0,0 +1,105 @@
+"""
+    loginpass.vk
+    ~~~~~~~~~~~~~~~~~~~
+
+    Authenticate using `VK <https://vk.com>`_.
+
+    Useful links:
+
+    - Register an application: https://vk.com/editapp?act=create
+    - API documentation: https://vk.com/dev/users.get
+
+    :copyright: (c) 2018 by Hsiaoming Yang
+    :license: AGPLv3+, see LICENSE for more details.
+"""
+
+import datetime
+from authlib.specs.oidc import UserInfo
+from authlib.common.urls import add_params_to_uri
+from ._core import OAuthBackend, map_profile_fields
+
+
+def vk_compliance_fix(session):
+    def _add_extra_info(url, headers, body):
+        params = {'v': '5.80', 'fields': 'sex,bdate,has_photo,photo_max_orig,site,screen_name'}
+        url = add_params_to_uri(url, params)
+        return url, headers, body
+
+    def _token_response(resp):
+        token = resp.json()
+        token['token_type'] = 'Bearer'
+        resp.json = lambda: token
+        return resp
+
+    session.register_compliance_hook('protected_request', _add_extra_info)
+    session.register_compliance_hook('access_token_response', _token_response)
+
+
+class VK(OAuthBackend):
+    OAUTH_TYPE = '2.0'
+    OAUTH_NAME = 'vk'
+    OAUTH_CONFIG = {
+        'api_base_url': 'https://api.vk.com/method/',
+        'access_token_url': 'https://oauth.vk.com/access_token',
+        'authorize_url': 'https://oauth.vk.com/authorize',
+        'client_kwargs': {
+            'token_placement': 'uri',
+            'token_endpoint_auth_method': 'client_secret_post',
+            'scope': 'email'
+        },
+        'compliance_fix': vk_compliance_fix
+    }
+
+    def profile(self, **kwargs):
```

Comment:
```python
def profile(self, token=None, **kwargs):
    params = {}
    if token and 'email' in token:
        params['email'] = token
```

I think this would be better.